lib/signet/oauth_2/client.rb

```diff
@@ -823,6 +828,25 @@ def expires_at= new_expires_at
         @expires_at = normalize_timestamp new_expires_at
       end
 
+      ##
+      # Returns the scopes granted by the authorization server.
+      #
+      # @return [Array] The scope of access returned by the authorization server.
+      def granted_scopes
+        @granted_scopes
+      end
+
+      ##
+      # Sets the scopes returned by authorization server for this client.
+      #
+      # @param [Array] new_granted_scopes
```

Comment:
If you want to support both Array and String (see your docs below), you should include both types here.

Also, do we need to support setting it back to `nil`?

Reply:
Thanks for pointing this out. This was a miss from my end.
The auth server returns a space delimited string, so I thought of supporting only `String` for now.
And yes, we can set it back to `nil`.